spaceship/lib/spaceship/portal/portal_client.rb

```diff
@@ -439,7 +439,11 @@ def devices(mac: false, include_disabled: false)
           sort: 'name=asc',
           includeRemovedDevices: include_disabled
         })
-        parse_response(r, 'devices')
+        result = parse_response(r, 'devices')
+
+        csrf_cache[Spaceship::Device] = self.csrf_tokens
```

Comment:
It's probably not in your code, but there's a smell here in that line 444 has to go between 442 and 446, even tough the local variable `result` isn't used in 444. I assume `self.csrf_tokens` is being populated by the call to `parse_response`, but it just seems a bit odd.

Reply:
That is correct they populate the `csrf_tokens` under the covers in `parse_response`.  I agree that it is odd.